### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by adaamko**
@recski  added expand and substitute functionality to TextTo4lang, also introduced a Fourlang class

### CODE REVIEW

**Review by recski — CHANGES_REQUESTED** [review]

File: tuw_nlp/grammar/text_to_4lang.py [hunk 1]
```diff
@@ -6,29 +6,72 @@
...
+            return
+
+        nodes = [node for node in graph.G.nodes(data=True)]
+        for d_node, node_data in nodes:
+            if all(("expanded", "substituted")) not in node_data:
+                node = graph.d_clean(node_data["name"]).split('_')[0]
+                if(node not in self.lexicon.stopwords or d_node == graph.root):
+                    definition = self.lexicon.get_definition(node)
+                    if definition:
+                        sen = self.nlp(definition).sentences[0]
```
[the remark this answers is not in the document]
Comment (adaamko) [reply tuw_nlp/grammar/text_to_4lang.py]: Should it be really cached? It is essentially just a dictionary lookup and the rest is already cached (ud parsing, UD_FL conversion)

[the remark this answers is not in the document]
Comment (recski) [reply tuw_nlp/grammar/text_to_4lang.py]: @adaamko you are right, sorry, if all steps are cached then there's no need. (But for that the self.nlp should also be a CachedPipeline, right?)

**Review by recski — APPROVED** [review]

File: tuw_nlp/grammar/text_to_4lang.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (adaamko) [reply tuw_nlp/grammar/text_to_4lang.py]: @recski yes self.nlp should be CachedPipeline, but we initialize it like that in line 28:
`        self.nlp = CachedStanzaPipeline(nlp, nlp_cache)
`

[the remark this answers is not in the document]
Comment (recski) [reply tuw_nlp/grammar/text_to_4lang.py]: excellent, thanks!

**Pull request merged**